src/qibo/models/evolution.py

```diff
@@ -280,17 +280,19 @@ def minimize(self, initial_parameters, method="BFGS", options=None, messages=Fal
                 optimization.
         """
         self.opt_messages = messages
-        if method == "sgd":
+        if isinstance(opt, optimizers.gradient_based.TensorflowSGD) == "sgd":
```

Comment:
```suggestion
        if isinstance(opt, optimizers.gradient_based.TensorflowSGD):
```
Right, or am I missing something?